src/main/java/org/folio/inventory/instanceingress/handler/CreateInstanceIngressEventHandler.java

```diff
@@ -0,0 +1,240 @@
+import java.util.Optional;
+import java.util.concurrent.CompletableFuture;
+import org.apache.commons.lang3.StringUtils;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.HttpStatus;
+import org.folio.MappingMetadataDto;
+import org.folio.inventory.common.Context;
+import org.folio.inventory.dataimport.cache.MappingMetadataCache;
+import org.folio.inventory.dataimport.handlers.actions.CreateInstanceEventHandler;
+import org.folio.inventory.dataimport.handlers.actions.PrecedingSucceedingTitlesHelper;
+import org.folio.inventory.dataimport.handlers.matching.util.EventHandlingUtil;
+import org.folio.inventory.dataimport.util.AdditionalFieldsUtil;
+import org.folio.inventory.dataimport.util.ValidationUtil;
+import org.folio.inventory.domain.instances.Instance;
+import org.folio.inventory.domain.instances.InstanceCollection;
+import org.folio.inventory.instanceingress.InstanceIngressEventConsumer;
+import org.folio.inventory.services.IdStorageService;
+import org.folio.inventory.storage.Storage;
+import org.folio.processing.exceptions.EventProcessingException;
+import org.folio.processing.mapping.defaultmapper.RecordMapper;
+import org.folio.processing.mapping.defaultmapper.RecordMapperBuilder;
+import org.folio.processing.mapping.defaultmapper.processor.parameters.MappingParameters;
+import org.folio.rest.jaxrs.model.InstanceIngressEvent;
+import org.folio.rest.jaxrs.model.ParsedRecord;
+import org.folio.rest.jaxrs.model.RawRecord;
+import org.folio.rest.jaxrs.model.Record;
+import org.folio.rest.jaxrs.model.Snapshot;
+
+public class CreateInstanceIngressEventHandler extends CreateInstanceEventHandler implements InstanceIngressEventHandler {
+
+  private static final Logger LOGGER = LogManager.getLogger(CreateInstanceIngressEventHandler.class);
+  private static final String BIBFRAME = " (bibframe)";
+  private final Context context;
+  private final InstanceCollection instanceCollection;
+
+  public CreateInstanceIngressEventHandler(PrecedingSucceedingTitlesHelper precedingSucceedingTitlesHelper,
+                                           MappingMetadataCache mappingMetadataCache,
+                                           IdStorageService idStorageService,
+                                           HttpClient httpClient,
+                                           Context context,
+                                           Storage storage) {
+    super(storage, precedingSucceedingTitlesHelper, mappingMetadataCache, idStorageService, null, httpClient);
+    this.context = context;
+    this.instanceCollection = storage.getInstanceCollection(context);
+  }
+
+  @Override
+  public CompletableFuture<InstanceIngressEvent> handle(InstanceIngressEvent event) {
+    try {
+      LOGGER.info("Processing InstanceIngressEvent with id '{}' for instance creation", event.getId());
+      if (isNull(event.getEventPayload()) || isNull(event.getEventPayload().getSourceRecordObject())) {
```

Comment:
suggest to move it to methods to separate business from techical
`if (isEmptyEvent(event))` {
    return createError(event.getId)
}

Reply:
sure, changed, thank you!